context/app/static/js/components/searchPage/MetadataMenu/MetadataMenu.jsx

```diff
@@ -11,43 +10,47 @@ import { useStore } from 'js/shared-styles/dropdowns/DropdownMenuProvider/store'
 import DropdownMenu from 'js/shared-styles/dropdowns/DropdownMenu';
 import { StyledDropdownMenuButton, StyledLink, StyledInfoIcon } from './style';
 
-function MetadataMenu({ type, analyticsCategory }) {
-  const lcPluralType = `${type.toLowerCase()}s`;
-  const allResultsUUIDs = useSearchViewStore((state) => state.allResultsUUIDs);
-
-  const { closeMenu } = useStore();
-  // eslint-disable-next-line consistent-return
-  async function fetchAndDownloadTSV() {
-    const response = await fetch(`/metadata/v0/${lcPluralType}.tsv`, {
+function makeDownloader({ urlPath, allResultsUUIDs, closeMenu, analyticsCategory }) {
+  async function fetchAndDownload() {
+    const response = await fetch(urlPath, {
       method: 'POST',
       body: JSON.stringify({ uuids: allResultsUUIDs }),
       headers: {
         'Content-Type': 'application/json',
       },
     });
     if (!response.ok) {
-      console.error('Metadata TSV Failed', response);
+      console.error('Download failed', response);
       closeMenu();
       return;
     }
     const results = await response.blob();
+    const headers = Object.fromEntries(Array(...response.headers));
+    const name = headers['content-disposition'].split('=')[1];
+    const mime = headers['content-type'];
 
-    const downloadUrl = createDownloadUrl(results, 'text/tab-separated-values');
-    // need to create link to set file name
+    const downloadUrl = createDownloadUrl(results, mime);
     const tempLink = document.createElement('a');
     tempLink.href = downloadUrl;
-    const timestamp = format(new Date(), 'yyyy-MM-dd_HH-mm-ss');
-    tempLink.download = `hubmap-${lcPluralType}-metadata-${timestamp}.tsv`;
+    tempLink.download = name;
     tempLink.click();
 
     ReactGA.event({
       category: analyticsCategory,
-      action: `Download Metadata`,
-      label: type,
+      action: `Download ${mime}`,
+      label: urlPath.split('/').pop(),
     });
```

Comment:
@tsliaw if this is merged as is the google analytics event action would be changed from 'Download Metadata' to 'Download text/tab-separated-values; charset=utf-8` moving forward. @mccalluc what would the mime for the notebook be?

Reply:
`application/x-ipynb+json` ... but if that's ugly, we could pass in an additional parameter: I was trying to keep the function signature simple, but if we want the stats to be continuous, it wouldn't be hard to pass in human-readable names. Let me know.